lib/rides/models/ride.dart

```diff
@@ -173,6 +173,29 @@ class Ride extends Trip {
   String toString() {
     return 'Ride{id: $id, in: $driveId, from: $start at $startTime, to: $end at $endTime, by: $riderId}';
   }
+
+  Ride copyWith({int? id, RideStatus? status}) {
```

Comment:
should we add a copyWith for every Model?

Reply:
Theoretically, yes. But as long as it's not needed, I would not do it (see PR description). Same with the attributes that can be changed. I only did those that are currently needed, furthers should be added on demand.